pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java

```diff
@@ -127,6 +163,22 @@ public String submitTask(@Nonnull PinotTaskConfig pinotTaskConfig, @Nonnull Stri
   }
 
   /**
+   * Get all tasks for the given task type.
+   *
+   * @param taskType Task type
+   * @return Set of task names
+   */
+  @Nonnull
+  public Set<String> getTasks(@Nonnull String taskType) {
```

Comment:
For now, I will make everything synchronized. If there are performance issues, will investigate later.